internal/file/tar_file_traversal.go

```diff
@@ -1,27 +1,52 @@
 package file
 
 import (
+	"context"
 	"fmt"
 	"os"
 	"path/filepath"
 
 	"github.com/bmatcuk/doublestar/v4"
+	"github.com/mholt/archives"
 
-	"github.com/anchore/archiver/v3"
+	"github.com/anchore/syft/internal/log"
 )
 
+// TraverseFilesInTar enumerates all paths stored within a tar archive using the visitor pattern.
+func TraverseFilesInTar(ctx context.Context, archivePath string, visitor archives.FileHandler) error {
+	tarReader, err := os.Open(archivePath)
+	if err != nil {
+		return fmt.Errorf("unable to open tar archive (%s): %w", archivePath, err)
+	}
+	defer func() {
+		if err := tarReader.Close(); err != nil {
+			log.Errorf("unable to close tar archive (%s): %+v", archivePath, err)
```

Comment:
This should not be an Error log, probably Debug. also there's an `internal.CloseAndLogError`